Damage indicator issue

**Describe the bug**
[ERROR] addons/ttt2-master/gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua:56: bad argument #1 to 'SetMaterial' (IMaterial expected, got nil)
1. SetMaterial - [C]:-1
2. unknown - addons/ttt2-master/gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua:56

This error happend with new user in my server, the client-fix of this problem is go to f1>damage indicator>force change .png image
It like "SetMaterial" receives "nill" for some reason, a check should be added to avoid this problem
Or maybe "SetMaterial" is setted "nill" before being initialized(?)

--- a/gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua
+++ b/gamemodes/terrortown/gamemode/client/cl_damage_indicator.lua
@@ -7,7 +7,7 @@ DMGINDICATOR.themes = {}
 -- SET UP CONVARS
 DMGINDICATOR.cv = {}
 DMGINDICATOR.cv.enable = CreateClientConVar("ttt_dmgindicator_enable", "1", true)
-DMGINDICATOR.cv.mode = CreateClientConVar("ttt_dmgindicator_mode", "Default", true)
+DMGINDICATOR.cv.mode = CreateClientConVar("ttt_dmgindicator_mode", "default", true)
 DMGINDICATOR.cv.duration = CreateClientConVar("ttt_dmgindicator_duration", "1.5", true)
 DMGINDICATOR.cv.maxdamage = CreateClientConVar("ttt_dmgindicator_maxdamage", "50.0", true)
 DMGINDICATOR.cv.maxalpha = CreateClientConVar("ttt_dmgindicator_maxalpha", "255", true)
@@ -57,4 +57,3 @@ function GM:HUDPaintBackground()
 		surface.DrawTexturedRect(0, 0, ScrW(), ScrH())
 	end
 end
-

--- a/gamemodes/terrortown/gamemode/client/cl_help.lua
+++ b/gamemodes/terrortown/gamemode/client/cl_help.lua
@@ -406,9 +406,6 @@ function HELPSCRN:CreateInterfaceSettings(parent)
 	cb = form:CheckBox(GetTranslation("disable_overheadicons"), "ttt2_disable_overheadicons")
 	cb:SetTooltip(GetTranslation("disable_overheadicons_tip"))
 
-	cb = form:CheckBox(GetTranslation("disable_doubletap_sprint"), "ttt2_disable_doubletap_sprint")
-	cb:SetTooltip(GetTranslation("disable_doubletap_sprint_tip"))
-
 	form:Dock(FILL)
 end
 
@@ -586,6 +583,9 @@ function HELPSCRN:CreateGameplaySettings(parent)
 
 	cb = form:CheckBox(GetTranslation("doubletap_sprint_anykey"), "ttt2_doubletap_sprint_anykey")
 	cb:SetTooltip(GetTranslation("doubletap_sprint_anykey_tip"))
+
+	cb = form:CheckBox(GetTranslation("disable_doubletap_sprint"), "ttt2_disable_doubletap_sprint")
+	cb:SetTooltip(GetTranslation("disable_doubletap_sprint_tip"))
 
 	-- TODO what is the following reason?
 	-- For some reason this one defaulted to on, unlike other checkboxes, so

--- a/gamemodes/terrortown/gamemode/client/cl_help.lua
+++ b/gamemodes/terrortown/gamemode/client/cl_help.lua
@@ -409,6 +409,9 @@ function HELPSCRN:CreateInterfaceSettings(parent)
 	cb = form:CheckBox(GetTranslation("disable_overheadicons"), "ttt2_disable_overheadicons")
 	cb:SetTooltip(GetTranslation("disable_overheadicons_tip"))
 
+	cb = form:CheckBox(GetTranslation("disable_doubletap_sprint"), "ttt2_disable_doubletap_sprint")
+	cb:SetTooltip(GetTranslation("disable_doubletap_sprint_tip"))
+
 	form:Dock(FILL)
 end
 

--- a/gamemodes/terrortown/gamemode/shared/lang/english.lua
+++ b/gamemodes/terrortown/gamemode/shared/lang/english.lua
@@ -1240,3 +1240,6 @@ L.disable_overheadicons_tip = "Disables the role icons over the players' heads (
 -- 2020-01-04
 L.doubletap_sprint_anykey = "Continue double tap sprinting until you stop moving"
 L.doubletap_sprint_anykey_tip = "You will keep sprinting as long as you keep moving"
+
+L.disable_doubletap_sprint = "Disable double tap sprinting"
+L.disable_doubletap_sprint_tip = "Double tapping a movement key will no longer cause you to sprint"

--- a/gamemodes/terrortown/gamemode/shared/lang/english.lua
+++ b/gamemodes/terrortown/gamemode/shared/lang/english.lua
@@ -1236,3 +1236,6 @@ L.disable_spectatorsoutline_tip = "Disables the outlines around objects controll
 
 L.disable_overheadicons = "Disable role icons"
 L.disable_overheadicons_tip = "Disables the role icons over the players' heads (+Performance)"
+
+L.disable_doubletap_sprint = "Disable doubletap Sprint"
+L.disable_doubletap_sprint_tip = "Disables Sprinting when double tapping the forward-move Key"

--- a/gamemodes/terrortown/gamemode/shared/lang/german.lua
+++ b/gamemodes/terrortown/gamemode/shared/lang/german.lua
@@ -1226,3 +1226,6 @@ L.disable_spectatorsoutline_tip = "Deaktiviert die Umrandungen um ein Objekt, in
 
 L.disable_overheadicons = "Deaktiviere Rollenicons"
 L.disable_overheadicons_tip = "Deaktiviert die Rollenicons über den Köpfen der Spieler (+Performance)"
+
+L.disable_doubletap_sprint = "Deaktiviere 'doubletap' Sprinten"
+L.disable_doubletap_sprint_tip = "Deaktiviert das Sprinten wenn die Nach-Vorne-Gehen Taste doppelt gedrückt wird"

--- a/gamemodes/terrortown/gamemode/shared/lang/german.lua
+++ b/gamemodes/terrortown/gamemode/shared/lang/german.lua
@@ -1230,3 +1230,6 @@ L.disable_overheadicons_tip = "Deaktiviert die Rollenicons über den Köpfen der
 -- 2020-01-04
 L.doubletap_sprint_anykey = "Setze Double-Tap Sprinten fort, bis du stehen bleibst"
 L.doubletap_sprint_anykey_tip = "So lange du dich weiterhin bewegst, wird das Sprinten fortgesetzt"
+
+L.disable_doubletap_sprint = "Deaktiviere Double-Tap Sprinten"
+L.disable_doubletap_sprint_tip = "Deaktiviert das Auslösen des Sprinten durch Double-Tap einer Bewegungstaste"

--- a/gamemodes/terrortown/gamemode/shared/sh_sprint.lua
+++ b/gamemodes/terrortown/gamemode/shared/sh_sprint.lua
@@ -47,8 +47,8 @@ if SERVER then
 	end)
 else
 	-- The helptext can't be changed once the convar was created, so we go with english, since it is probably the most common lang
-	local doubletap_sprint_anykey = CreateClientConVar("ttt2_doubletap_sprint_anykey", 0, true, false, LANG.GetTranslationFromLanguage("disable_doubletap_sprint", "english"), 0, 1)
-	local disable_doubletap_sprint = CreateClientConVar("ttt2_disable_doubletap_sprint", "0", true, false, LANG.GetTranslationFromLanguage("doubletap_sprint_anykey", "english"), 0, 1)
+	local doubletap_sprint_anykey = CreateClientConVar("ttt2_doubletap_sprint_anykey", 0, true, false, LANG.GetTranslationFromLanguage("doubletap_sprint_anykey", "english"), 0, 1)
+	local disable_doubletap_sprint = CreateClientConVar("ttt2_disable_doubletap_sprint", "0", true, false, LANG.GetTranslationFromLanguage("disable_doubletap_sprint", "english"), 0, 1)
 	local lastPress = 0
 	local lastPressedMoveKey = nil
 

--- a/gamemodes/terrortown/gamemode/shared/sh_sprint.lua
+++ b/gamemodes/terrortown/gamemode/shared/sh_sprint.lua
@@ -47,7 +47,8 @@ if SERVER then
 	end)
 else
 	-- The helptext can't be changed once the convar was created, so we go with english, since it is probably the most common lang
-	local doubletap_sprint_anykey = CreateClientConVar("ttt2_doubletap_sprint_anykey", 0, true, false, LANG.GetTranslationFromLanguage("doubletap_sprint_anykey", "english"), 0, 1)
+	local doubletap_sprint_anykey = CreateClientConVar("ttt2_doubletap_sprint_anykey", 0, true, false, LANG.GetTranslationFromLanguage("disable_doubletap_sprint", "english"), 0, 1)
+	local disable_doubletap_sprint = CreateClientConVar("ttt2_disable_doubletap_sprint", "0", true, false, LANG.GetTranslationFromLanguage("doubletap_sprint_anykey", "english"), 0, 1)
 	local lastPress = 0
 	local lastPressedMoveKey = nil
 
@@ -69,7 +70,7 @@ else
 	end
 
 	hook.Add("KeyPress", "TTT2DoublePressSprint", function(ply, key)
-		if not IsFirstTimePredicted() or ply.isSprinting then return end
+		if not IsFirstTimePredicted() or ply.isSprinting or disable_doubletap_sprint:GetBool() then return end
 		if key ~= IN_FORWARD and key ~= IN_BACK and key ~= IN_MOVERIGHT and key ~= IN_MOVELEFT then return end
 
 		local time = CurTime()

--- a/gamemodes/terrortown/gamemode/shared/sh_sprint.lua
+++ b/gamemodes/terrortown/gamemode/shared/sh_sprint.lua
@@ -46,6 +46,9 @@ if SERVER then
 		ply.isSprinting = bool
 	end)
 else
+
+	local disable_doubletap_sprint = CreateClientConVar("ttt2_disable_doubletap_sprint", "0", true, true)
+
 	local lastPress = 0
 
 	local function PlayerSprint(trySprinting, bind)
@@ -66,7 +69,7 @@ else
 	end
 
 	hook.Add("KeyPress", "TTT2DoublePressSprint", function(ply, key)
-		if not IsFirstTimePredicted() then return end
+		if not IsFirstTimePredicted() or disable_doubletap_sprint:GetBool() then return end
 		if key == IN_FORWARD and not ply.isSprinting then
 			local time = CurTime()
 

